wallet/src/account/handle.rs

```diff
@@ -203,34 +206,39 @@ impl AccountHandle {
             }
             outputs.push(output.clone());
         }
+
         Ok(outputs)
     }
 
     /// Returns all incoming transactions of the account
     pub async fn incoming_transactions(&self) -> Result<HashMap<TransactionId, Transaction>> {
-        let account = self.read().await;
-        Ok(account.incoming_transactions.clone())
+        Ok(self.read().await.incoming_transactions.clone())
     }
 
     /// Returns all transactions of the account
     pub async fn transactions(&self) -> Result<Vec<Transaction>> {
-        let account = self.read().await;
         let mut transactions = Vec::new();
-        for transaction in account.transactions.values() {
+
+        #[allow(clippy::significant_drop_in_scrutinee)]
+        for transaction in self.read().await.transactions.values() {
             transactions.push(transaction.clone());
         }
+
         Ok(transactions)
```

Comment:
This can be just `Ok(self.read().await.transactions.values().cloned().collect())`